src/pudl/metadata/constants.py

```diff
@@ -30,7 +31,10 @@
 FIELD_DTYPES_SQL: dict[str, sa.sql.visitors.VisitableType] = {
     "boolean": sa.Boolean,
     "date": sa.Date,
-    "datetime": sa.DateTime,
+    # Ensure SQLite's string representation of datetime uses only whole seconds:
+    "datetime": SQLITE_DATETIME(
+        storage_format="%(year)04d-%(month)02d-%(day)02d %(hour)02d:%(minute)02d:%(second)02d"
+    ),
     "integer": sa.Integer,
     "number": sa.Float,
     "string": sa.Text,
```

Comment:
This lets the datetime pseudo-type check work, and removes artificial precision in the datetime fields.

I think that we have enough SQLite specific stuff elsewhere in the system that our `to_sql()` is functionally `to_sqlite()` and so having an SQLite specific dtype in here (rather than the SQLAlchemy generics) isn't any more constrained than we already are.

If we want to be able to write into other DBs I think we'll probably want dialect-specific schema generation methods, like `to_sqlite()`, `to_bigquery()` and `to_postgres()`